proposals/texture-buffers.md

```diff
@@ -0,0 +1,405 @@
+
+
+## Extension Names
+
+Add `'texture_buffers'` as a new extension name.
+
+
+## Enable Extensions
+
+[[Add new table entry to *Enable-extensions*]]
+
+| WGSL enable-extension | WebGPU GPUFeatureName | Description |
+| --------------------- | --------------------- | ----------- |
+| **texture_buffers**   | `"texture-buffers"`   | The `texture_buffer` type is valid to use in the WGSL module. Otherwise, using `texture_buffer` will result in a shader-creation error. |
+
+
+## Texture Buffer Types
+
+[[New subsection of **Texture and Sampler Types**]]
+
+A **texture buffer** supports accessing texels stored in a 1D buffer using texture load and store functions.
+Use of this type requires enabling the `texture_buffer` extension.
+
+A texture buffer type must be parameterized by one of the [texel formats](https://w3.org/TR/WGSL/#texel-formats) for storage textures.
+The texel format determines the conversion function as specified in [Texel Formats](https://w3.org/TR/WGSL/#texel-formats).
+
+For a `textureStore` operation, the inverse of the conversion function is used to convert the shader value to the stored texel.
+
+| Type                                   | Description                    |
+| -------------------------------------- | ------------------------------ |
+| **texture_buffer**<_Format_, _Access_> | A texture buffer type that accesses buffer data using texture functions. |
+
+- _Format_ must be an enumerant for one of the texel formats for storage textures
+- _Access_ must be `read` or `read_write`
+
+Writes to texture buffers are visible to the same invocation, and can be synchronized with other invocations from the same workgroup using a `textureBarrier`.
+
+
+## Restrictions on Functions
+
+Add `texture_buffer` to the list of valid function parameter types.
+
+
+## Texture Built-in Functions
+
+[[Add new overloads]]
+
+| Parameterization               | Overload                 |
+| ------------------------------ | ------------------------ |
+| _AM_ is `read` or `read_write` | `@must_use fn textureDimensions(t : texture_buffer<F, AM>) -> u32` |
+| _C_ is `i32` or `u32`<br/>_AM_ is `read` or `read_write`<br/>_CF_ depends on the storage texel format _F_. [See the texel format table](https://w3.org/TR/WGSL/#storage-texel-formats) for the mapping of texel format to channel format. | `@must_use fn textureLoad(t : texture_buffer<F, AM>, coords : C) -> vec4<CF>` |
+| _C_ is `i32` or `u32`<br/>_CF_ depends on the storage texel format _F_. [See the texel format table](https://w3.org/TR/WGSL/#storage-texel-formats) for the mapping of texel format to channel format. | `@must_use fn textureStore(t : texture_buffer<F, read_write>, coords : C, value: vec4<CF>)` |
+
+
+# API
+
+
+## Limits
+
+| Limit name               | Type        | Limit class | Default                   |
+| ------------------------ | ----------- | ----------- | ------------------------- |
+| **maxTextureBufferSize** | `GPUSize64` | maximum     | 134217728 bytes (128 MiB) |
```

Comment:
Need to be clear the units are bytes.

Reply:
Looks clear to me already? Might need to drop it to 64 MiB for the Apple2 GPU family, unless we don't care about that family.